policy/runtime/runtime.go

```diff
@@ -290,10 +290,11 @@ func (t *Template) newEvaluator(mdl *model.Evaluator,
 				agg = &CollectAggregator{name: d.Name}
 			}
 			decs[i] = &decision{
-				name: d.Name,
-				slot: slot,
-				prg:  dec,
-				agg:  agg,
+				name:  d.Name,
+				slot:  slot,
+				prg:   dec,
+				agg:   agg,
+				field: d.Field,
```

Comment:
This is a very clever way to plumb through the field; however, I think it would be better to pack the field name (once validated) into the `Output` value for the decision and tease out the field id once a violation is reported:

The following code snippet is used to populate the instance validation error message:
https://github.com/google/cel-policy-templates-go/blob/ef8e886926b07ecde649f5189d780f2320b88ab2/policy/runtime/runtime.go#L171-L173

If you were to take a similar approach to embedding the `field` in the `Output` ast generated during template compilation, you could then perform lookups by field name within the context of reporting the violation. It feels a bit like cheating, but it keeps the aggregation / decision logic pure.

Reply:
done.
I thought about this approach but somehow got lost